Bowon/MemberList/MemberList/MemberListViewController.swift

```diff
@@ -0,0 +1,172 @@
+    }
+    
+    @objc func onTapAddBarButton(_:UIBarButtonItem){
+        let AddMemberVC = AddMemberViewController()
+        self.navigationController?.pushViewController(AddMemberVC, animated: true)
+    }
+}
+
+// MARK: - extension UITableViewDelegate,UITableViewDataSource
+extension MemberListViewController: UITableViewDelegate, UITableViewDataSource, MemberDetailDelegate{
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: MemberListViewCell.identifier, for: indexPath) as? MemberListViewCell else { return UITableViewCell() }
+        
+        guard indexPath.row < memberList.count else { return UITableViewCell() }
+
+        let data = memberList[(indexPath as NSIndexPath).row]
+        
+        cell.memberNameLabel.text = data.memberName
+        cell.memberPicturesImageView.image = UIImage(named:data.memberImage!)
+        cell.memberAddressLabel.text = data.memberAddress
+        
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return memberList.count
+    }
+    
+    //editing style
+    func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+        if editingStyle == .delete {
+            memberList.remove(at: (indexPath as NSIndexPath).row)
+            tableView.deleteRows(at: [indexPath], with: .fade)
+        } 
+//        else if editingStyle == .insert {
+//
+//        }
+    }
+    
+    func tableView(_ tableView: UITableView, titleForDeleteConfirmationButtonForRowAt indexPath: IndexPath) -> String? {
+        return "삭제"
+    }
+    
+    //cell 위치 움직이는 함수
+    func tableView(_ tableView: UITableView, moveRowAt fromIndexPath: IndexPath, to: IndexPath) {
+        let moveList = memberList[(fromIndexPath as NSIndexPath).row]
+        
+        memberList.remove(at: (fromIndexPath as NSIndexPath).row)
+        memberList.insert(moveList, at: (to as NSIndexPath).row)
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let memberData = memberList[indexPath.row]
+        
+        selectedMemberIndex = indexPath.row
+        let UpdateMemberVC = UpdateMemberViewController()
+                
+        UpdateMemberVC.UpdateMemberImageView.image = UIImage(named: memberData.memberImage!)
+        UpdateMemberVC.UpdateMemberNumberTextField.text = memberData.memberNumber
+        UpdateMemberVC.UpdateMemberNameTextField.text = memberData.memberName
+        UpdateMemberVC.UpdateMemberAgeTextField.text = memberData.memberAge
+        UpdateMemberVC.UpdateMemberAddressTextField.text = memberData.memberAddress
+        UpdateMemberVC.UpdateMemberPhoneNumberTextField.text = memberData.memberPhoneNumber
+        
+        self.navigationController?.pushViewController(UpdateMemberVC, animated: true)
+    }
```

Comment:
이렇게 해도 별 문제는 없겠지만,
해당 VC에서 너무 많은 로직을 담당하는 것 같아서
이런 데이터 주입은 `updateMemberVC`에서 따로 처리해주는게 더 나을 것 같아.

```suggestion
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let memberData = memberList[indexPath.row]
        
        selectedMemberIndex = indexPath.row
        let UpdateMemberVC = UpdateMemberViewController()
        UpdateMemberViewController(memberData)
        
        self.navigationController?.pushViewController(UpdateMemberVC, animated: true)
    }
```